hooks/mutations/delegation/useSendRequestToBeDelegate.ts

```diff
@@ -0,0 +1,16 @@
+import { useMutation } from "@tanstack/react-query";
+import { useHandleError } from "hooks";
+import { setDelegate } from "web3";
+
+export function useSendRequestToBeDelegate(errorType?: string) {
```

Comment:
Here the delegator is doing a setDelegate to eventually become a delegator right?
I think we should rename it:
```suggestion
export function useSendRequestToBeDelegator(errorType?: string) {
```

Reply:
What's happening here is that the user is asking someone else to be their delegate. So we could phrase it either as "asking someone to be your delegate" OR "asking someone to be their delegator" and either would be correct. I've chosen the former way here, but I'm reaaaally hoping that with some input from @smb2796 et al we'll change the terms here and we can do something a bit less confusing.

Reply:
So on the naming convention - agreed that using delegator and delegate is pretty confusing (especially from code perspective). I looked at Aave, ENS, Uniswap etc, and they all use delegate / delegator, but do not have this explicit acceptance from delegate side, so it does not confuse UX as much. A few options:
- Delegator & Representative
- Delegator and Voter
- Owner and delegate
- Tokenholder and delegate

My preference is probably the first because I think it works for all case (setTokenholder doesn't really make too much sense), but perhaps it is too long. Thoughts?

Reply:
Interesting that the others use delegator / delegate. Now that I've gotten more used to the terms, they've sort of grown on me. Maybe they aren't that confusing after all.

I really am not sure what I think there. 

@chrismaree @md0x @mrice32 do you guys maybe wanna weigh in here?

Reply:
As long as it's not too confusing from code perspective, I don't think it is for UX